### PR TITLE
VIEWS : Guardian Request Navigation

### PR DESCRIPTION
**Pull request by imjane5**

### CODE REVIEW

File: SCMS.Portal.Web/Models/Views/Foundations/GuardianRequestViews/Exceptions/InvalidGuardianRequestViewException.cs [hunk 1]
```diff
@@ -11,5 +11,11 @@ public class InvalidGuardianRequestViewException : Xeption
         public InvalidGuardianRequestViewException()
             : base("Invalid guardianView error occured, fix errors and try again.")
         { }
+
+        public InvalidGuardianRequestViewException(string parameterName, object parameterValue)
+           : base($"Invalid student view error occured. " +
```
Comment (ShriHumrudha) [inline SCMS.Portal.Web/Models/Views/Foundations/GuardianRequestViews/Exceptions/InvalidGuardianRequestViewException.cs:16]: guardian request

File: SCMS.Portal.Web/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewService.Validations.cs [hunk 2]
```diff
@@ -28,6 +28,18 @@ private void ValidateGuardianRequestViewOnAdd(GuardianRequestView guardianReques
...
+            if (IsInvalidRoute(route))
+            {
+                throw new InvalidGuardianRequestViewException(
+                    parameterName: "Route",
+                    parameterValue: route);
+            }
+        }
+
+        private bool IsInvalidRoute(string route) =>
+            string.IsNullOrWhiteSpace(route);
```
Comment (ShriHumrudha) [inline SCMS.Portal.Web/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewService.Validations.cs:42]: Single line space is required after the method

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace SCMS.Portal.Tests.Unit.Services.Views.Foundations.GuardianRequestViews
+{
+    public partial class GuardianRequestViewServiceTests
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public void ShouldThrowValidationExceptionOnNavigateIfRouteIsInvalidAndLogitAsync(
```
[the remark this answers is not in the document]
Comment (tharanisantosh) [reply SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs:18]: Comment Resolved

File: SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedGuardianRequestViewValidationException))),
+                        Times.Once);
+
+            this.navigationBrokerMock.Verify(broker =>
+                broker.NavigateTo(It.IsAny<string>()),
+                    Times.Never);
+
+            this.navigationBrokerMock.VerifyNoOtherCalls();
```
[the remark this answers is not in the document]
Comment (tharanisantosh) [reply SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs:46]: Comment Resolved

File: SCMS.Portal.Web/Models/Views/Foundations/GuardianRequestViews/Exceptions/InvalidGuardianRequestViewException.cs (hunk 1 above)
Comment (tharanisantosh) [reply SCMS.Portal.Web/Models/Views/Foundations/GuardianRequestViews/Exceptions/InvalidGuardianRequestViewException.cs:16]: Comment Resolved

File: SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public void ShouldThrowValidationExceptionOnNavigateIfRouteIsInvalidAndLogit(
+            string invalidRoute)
+        {
+            // given
+            var invalidGuardianRequestViewException =
+               new InvalidGuardianRequestViewException(
```
Comment (ShriHumrudha) [inline SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs:23]: Check the alignment here.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs (hunk 5 above)
Comment (ShriHumrudha) [reply SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs:23]: @imjane5 @tharanisantosh - This is pending.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs (hunk 5 above)
Comment (tharanisantosh) [reply SCMS.Portal.Tests.Unit/Services/Views/Foundations/GuardianRequestViews/GuardianRequestViewServiceTests.Validations.Navigate.cs]: Comment Resolved.

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**